gpkit/nomial_data.py

```diff
@@ -54,6 +45,24 @@ def __hash__(self):
             self._hashvalue = hash(tuple(zip(self.exps, self.cs)))
         return self._hashvalue
 
+    @classmethod
+    def fromnomials(cls, nomials):
+        """Construct a NomialData from an iterable of Signomial objects"""
+        nd = cls()  # use pass-through version of __init__
```

Comment:
hmm, what I like about this is that it's causing discussion about the right class design / inheritance structure.

It's both impressive and a bit awkward that Signomial and GeometricProgram can both inherit from NomialData. I think it's the right design -- it encodes cs and exps as the right building block. The awkwardness comes in the fact that GP needs a `map` too (i.e. `p_idxs`). And that's related to the fact that it makes sense to create a GP from nomials, but not so much a Signomial from nomials.